.github/workflows/dependabot-pnpm-lock.yml

```diff
@@ -0,0 +1,40 @@
+name: Update pnpm-lock.yaml for Dependabot PRs
+
+on:
+  pull_request_target:
+    types: [opened, synchronize]
+    paths:
+      - '**/package.json'
+  workflow_dispatch:
+
+jobs:
+  update-lockfile:
+    runs-on: ubuntu-latest
+    if: ${{ github.actor == 'dependabot[bot]' }}
+    permissions:
+      contents: write
+      pull-requests: write
+
+    steps:
+      - uses: actions/checkout@v4
```

Comment:
this is also done in the setup action, you can review how we do it in the auto-fix-lint action